crates/primitives-traits/src/transaction/mod.rs

```diff
@@ -1,28 +1,59 @@
-//! Transaction abstraction
+use core::{fmt::Debug, hash::Hash};
 
-pub mod signed;
-
-use alloc::fmt;
+use alloy_primitives::{TxKind, B256};
 
+use alloy_rlp::Encodable;
 use reth_codecs::Compact;
 use serde::{Deserialize, Serialize};
 
-/// Helper trait that unifies all behaviour required by transaction to support full node operations.
-pub trait FullTransaction: Transaction + Compact {}
-
-impl<T> FullTransaction for T where T: Transaction + Compact {}
```

Comment:
revert this, we don't want to require `Compact` be a super trait for `Transaction`